Source/AliveLibAO/Abe.cpp

```diff
@@ -7885,9 +7887,322 @@ void Abe::State_87_428FA0()
+                field_110_state = StoneStates::eUnknown_5;
+            }
+            break;
+        }
+        case StoneStates::eUnknown_3:
+        {
+            gConter_507728--;
+            if (!gConter_507728)
+            {
+                field_110_state = StoneStates::eUnknown_5;
+                ResourceManager::FreeResource_455550(gpDemoPlaybackRes_50772C);
+                auto pCircularFade = Make_Circular_Fade_447640(
+                    field_A8_xpos,
+                    field_AC_ypos,
+                    field_BC_sprite_scale,
+                    0,
+                    0
+                );
+
+                field_164_pCircularFade = pCircularFade;
+                if (field_10_anim.field_4_flags.Get(AnimFlags::eBit5_FlipX))
+                {
+                    field_164_pCircularFade->field_10_anim.field_4_flags.Set(AnimFlags::eBit5_FlipX);
+                }
+                else
+                {
+                    field_164_pCircularFade->field_10_anim.field_4_flags.Clear(AnimFlags::eBit5_FlipX);
+                }
+            }
+            break;
+        }
+        case StoneStates::eUnknown_4:
+        {
+            if (sBellSong_507724->field_14_bDone)
+            {
+                sBellSong_507724->field_6_flags.Set(Options::eDead_Bit3);
+                field_164_pCircularFade->Vsub_479FE0(
+                    0,
+                    0
+                );
+                field_110_state = StoneStates::eUnknown_5;
+            }
+            break;
+        }
+        case StoneStates::eUnknown_5:
+        {
+            if (field_164_pCircularFade->Vsub_47A4C0())
+            {
+                field_164_pCircularFade->field_6_flags.Set(Options::eDead_Bit3);
+                field_FC_current_motion = eAbeStates::State_89_HandstoneEnd_430E80;
+                field_164_pCircularFade = 0;
+                if (sAbeSound_507730)
+                {
+                    SND_Stop_Channels_Mask_4774A0(sAbeSound_507730);
+                    sAbeSound_507730 = 0;
+                }
+                ResourceManager::FreeResource_455550(
+                    ResourceManager::GetLoadedResource_4554F0(ResourceManager::Resource_Animation, 316, 0, 0)
+                );
+            }
+            break;
+        }
+        case StoneStates::eUnknown_6:
+        {
+            if (field_158_pDeathFadeout->field_6E_bDone)
+            {
+                if (sInputObject_5009E8.isHeld(0xF0))
+                {
+                    field_158_pDeathFadeout->Init_419E40(40, 1, 0, 8);
+                    field_110_state = StoneStates::eUnknown_7;
+                    SFX_Play_43AD70(107u, 90, 0);
+                }
+            }
+            break;
+        }
+        case StoneStates::eUnknown_7:
+        {
+            if (field_158_pDeathFadeout->field_6E_bDone)
+            {
+                auto camera = field_174_pathStone.dataHandstone.cameras[field_16E_cameraIdx];
```

Comment:
could be const? maybe use the actual type here as well since its not obvious what it is